Parse/src/main/java/com/parse/ParseFile.java

```diff
@@ -298,6 +369,10 @@ public void save() throws ParseException {
    * @return A Task that will be resolved when the save completes.
    */
   public Task<Void> saveInBackground(final ProgressCallback uploadProgressCallback) {
+    if (data != null && file != null) {
+      throw new IllegalArgumentException("File and data can not be set at the same time");
+    }
+
```

Comment:
It is impossible. I remembered you recommended me to add this checking in `ParseRESTFileCommand` constructor so I also add it here. Do we still need this?